restapi/src/main/java/io/stargate/web/docsapi/service/QueryExecutor.java

```diff
@@ -116,20 +212,37 @@ public QueryExecutor(DataStore dataStore) {
     }
   }
 
-  private Iterable<Accumulator> seeds(
-      BoundQuery query, ResultSet rs, List<Column> keyColumns, ExecutionContext context) {
+  private Accumulator toSeed(
+      DocProperty property, Comparator<DocProperty> comparator, List<Column> keyColumns) {
+    Row row = property.row();
+    String id = row.getString("key");
+    Builder<String> docKey = ImmutableList.builder();
+    for (Column c : keyColumns) {
+      docKey.add(Objects.requireNonNull(row.getString(c.name())));
+    }
+
+    return new Accumulator(id, comparator, docKey.build(), property);
+  }
+
+  private Iterable<DocProperty> properties(
```

Comment:
can you explain this a bit more to me please? also some kind of inline docs would be so helpful to understand what s going on, nobody will ever be able to understand this except the writer :)

Reply:
I'll add comments

Reply:
javadoc added